app/assets/javascripts/discourse/app/components/sidebar/anonymous/custom-sections.js

```diff
@@ -1,17 +1,29 @@
 import Component from "@glimmer/component";
 import { inject as service } from "@ember/service";
 import Section from "discourse/components/sidebar/user/section";
+import SystemSection from "discourse/components/sidebar/common/system-section";
 
 export default class SidebarAnonymousCustomSections extends Component {
   @service router;
   @service site;
+  @service siteSettings;
 
   get sections() {
     return this.site.anonymous_sidebar_sections?.map((section) => {
-      return new Section({
-        section,
-        router: this.router,
-      });
+      if (section.system) {
+        return new SystemSection({
+          section,
+          currentUser: this.currentUser,
+          router: this.router,
+          siteSettings: this.siteSettings,
+        });
+      } else {
+        return new Section({
+          section,
+          currentUser: this.currentUser,
+          router: this.router,
+        });
+      }
```

Comment:
```suggestion
      const klass = section.system ? SystemSection : Section

      return new klass({
          section,
          currentUser: this.currentUser,
          router: this.router,
          siteSettings: this.siteSettings,
      })
```

I think we can just simplify this by always passing site settings through.